core/domain/skill_jobs_one_off_test.py

```diff
@@ -329,51 +352,83 @@ def test_migration_job_skips_deleted_model(self):
-                'old_value': '', 'property_name': 'description'
-            }])
+            self.commit_model_instance_1.commit_cmds, self.invalid_commit_cmd)
+
         job_id = (
             skill_jobs_one_off.SkillCommitCmdMigrationOneOffJob.create_new())
         skill_jobs_one_off.SkillCommitCmdMigrationOneOffJob.enqueue(job_id)
         self.process_and_flush_pending_mapreduce_tasks()
-
-        output = skill_jobs_one_off.SkillCommitCmdMigrationOneOffJob.get_output(
-            job_id)
+        stringified_output = (
+            skill_jobs_one_off.SkillCommitCmdMigrationOneOffJob.get_output(
+                job_id))
+        eval_output = [
+            ast.literal_eval(stringified_item) for
+            stringified_item in stringified_output]
+        expected_output = [
+            [
+                'Commit Commands Updated-SkillCommitLogEntryModel',
+                ['skill-skill_id-2']
+            ]
+        ]
+
+        self.assertItemsEqual(eval_output, expected_output)
         self.assertEqual(
-            output, ['[u\'Commit Commands Updated\', [u\'skill-skill_id-2\']]'])
-        self.assertEqual(
-            self.model_instance_0.commit_cmds, [{'cmd': 'create_new'}])
-        self.model_instance_1 = (
+            self.commit_model_instance_0.commit_cmds, self.default_commit_cmd0)
+        self.commit_model_instance_1 = (
             skill_models.SkillCommitLogEntryModel.get_by_id(
                 'skill-skill_id-2'))
         self.assertEqual(
-            self.model_instance_1.commit_cmds,
-            [{
-                'difficulty': 'Easy', 'cmd': 'update_rubrics',
-                'explanations': ['New explanation']
-            }, {
-                'cmd': 'update_skill_property', 'new_value': 'Test description',
-                'old_value': '', 'property_name': 'description'
-            }])
+            self.commit_model_instance_1.commit_cmds,
+            self.default_commit_cmd1)
+
+    def test_migration_job_updates_invalid_command_of_metadata_model(self):
+        self.metadata_model_instance_1.commit_cmds = self.invalid_commit_cmd
+        self.metadata_model_instance_1.update_timestamps()
+        self.metadata_model_instance_1.put()
+
+        self.assertEqual(
+            self.metadata_model_instance_0.commit_cmds,
+            self.default_commit_cmd0)
+        self.assertEqual(
+            self.metadata_model_instance_1.commit_cmds,
+            self.invalid_commit_cmd)
+
+        job_id = (
+            skill_jobs_one_off.SkillCommitCmdMigrationOneOffJob.create_new())
+        skill_jobs_one_off.SkillCommitCmdMigrationOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_mapreduce_tasks()
+        stringified_output = (
+            skill_jobs_one_off.SkillCommitCmdMigrationOneOffJob.get_output(
+                job_id))
+        eval_output = [
+            ast.literal_eval(stringified_item) for
+            stringified_item in stringified_output]
```

Comment:
Move this code to separate private methods `_run_one_off_job` and call it from places (we already use this approach in other files).

Reply:
Done.